Source/CourseContentPageViewController.swift

```diff
@@ -111,6 +115,18 @@ public class CourseContentPageViewController : UIPageViewController, UIPageViewC
         addObservers()
         
         ChromeCastManager.shared.removeChromeCastButton(from: self, force: true)
+        addRestrictedViewToPagination()
+    }
+    
+    // This is to restrict the pagination for bottom bar of player to make player progress slider smooth
+    private func addRestrictedViewToPagination() {
+        for view in view.subviews {
+            if view is UIScrollView {
+                let scrollView = view as? UIScrollView
+                scrollViewPanGestureRecognzier.delegate = self
+                scrollView?.addGestureRecognizer(scrollViewPanGestureRecognzier)
+            }
+        }
```

Comment:
These lines can be simplified as
```suggestion
        for view in view.subviews {
            if let scrollView = view as? UIScrollView {
                scrollViewPanGestureRecognzier.delegate = self
                scrollView.addGestureRecognizer(scrollViewPanGestureRecognzier)
            }
        }
```

Reply:
done